oak-auth-external/src/main/java/org/apache/jackrabbit/oak/spi/security/authentication/external/impl/principal/ExternalGroupPrincipalProvider.java

```diff
@@ -101,6 +102,7 @@
 class ExternalGroupPrincipalProvider implements PrincipalProvider, ExternalIdentityConstants, DynamicMembershipProvider {
 
     private static final Logger log = LoggerFactory.getLogger(ExternalGroupPrincipalProvider.class);
+    static final String CACHE_PRINCIPAL_NAMES = "externalPrincipalNames";
```

Comment:
i would recommend to use "rep:" prefix for the property to mark it as oak-system maintained property

Reply:
Done, instead of adding `rep:` in front I renamed it as there are some restrictions to that property definition.

Reply:
@Amoratinos , how do you mean? sorry you lost me here.... why not adding the rep: prefix?

Reply:
`rep:externalPrincipalNames` is already taken use, right? so, i would avoid re-using the same name for the caching prop.... specially since it is not totally accurate. the cache will also contain local group names, which external groups are member of, no?
let's think about another name and add rep prefix. this is important to mark it as oak-internal property.
e.g.
- rep:groupPrincipalNamesForExternal
- rep:groupPrincipalNamesExternal
- ....
wdyt?

Reply:
I've renamed to `rep:externalCachedPrincipalNames` if that works